src/graph_utils.rs

```diff
@@ -64,10 +70,61 @@ pub fn find_bridges(nodes: &[Node], threshold_adjustment: f64) -> HashMap<usize,
     bridges
 }
 
+pub fn construct_graph(nodes: &[Node]) -> Graph<IpAddr> {
+    let mut graph = Graph::new();
+
+    for i in 0..nodes.len() {
+        for j in 0..nodes[i].connections.len() {
+            let edge = Edge::new(
+                IpAddr::from_str(nodes[i].ip.as_str()).expect(ERR_PARSE_IP),
```

Comment:
one improvement could be to create this IpAddr between lines 76 and 77 since this is a static value between all j iterations and then only copy it here

Reply:
That's true 👍 Fixed.